test/EntityFramework/FunctionalTests/PlanCompiler/LimitExpressionTests.cs

```diff
@@ -834,6 +836,42 @@ orderby building.Address.ZipCode
         }
 
         [Fact]
+        public void Limit_ComplexModel_OrderBy_Skip_without_rownumber()
+        {
+            var log = new StringWriter();
+            using (var context = new AdvancedPatternsMasterContext())
+            {
+                var previousFlag = SqlProviderServices.UseOrderByRowNumberForQueriesWithOffset;
+                try
+                {
+                    SqlProviderServices.UseOrderByRowNumberForQueriesWithOffset = false;
```

Comment:
Re the brittle test. I would prefer not to add it. As long as all existing tests and new tests pass with the new behavior I think it is fine.